storm/src/main/storm/mesos/MesosNimbus.java

```diff
@@ -335,6 +382,92 @@ public boolean isHostAccepted(String hostname) {
+
+    Map<String, AggregatedOffers> aggregatedOffersPerNode = MesosCommon.getAggregatedOffersPerNode(_offers);
+
+    String supervisorStormLocalDir = getStormLocalDirForWorkers();
+    String configUri = getFullConfigUri();
+    String logviewerCommand = String.format(
+                    "cp storm.yaml storm-mesos*/conf" +
+                    " && cd storm-mesos*" +
+                    " && python bin/storm logviewer" +
+                    " -c storm.local.dir=%s", supervisorStormLocalDir);
+
+    /**
+     *  Go through each existing supervisor and:
+     *    1. Check ZK if logviewer already exists on worker host, if not then continue
+     *    2. Look in the aggregratedOffersPerNode and check the node for offers, if there are offers then continue
+     *    3. Create a TaskInfo for the logviewer corresponding to the correct Offers and launch the task
+     *    4. Update ZK to reflect the existence of new logviewer on worker host
+     */
+
+    for (SupervisorDetails supervisor : existingSupervisors) {
+      List<TaskInfo> logviewerTask = new ArrayList<TaskInfo>();
+      LOG.info("launchLogviewer: Supervisor ID: {}", supervisor.getId());
+
+      String nodeId = supervisor.getId().split("\\" + DEFAULT_MESOS_COMPONENT_NAME_DELIMITER)[0];
+
+      if (_zkClient.nodeExists(String.format("%s/%s", _logviewerZkDir, nodeId))) {
+        LOG.info("launchLogviewer: Logviewer already exists on this host: {}", nodeId);
+        continue;
+      }
+
+      AggregatedOffers aggregatedOffers = aggregatedOffersPerNode.get(nodeId);
+      if (aggregatedOffers == null) {
+        LOG.info("launchLogviewer: No offers for this host: {}", nodeId);
+        continue;
+      }
+
+      List<OfferID> offerIDList = aggregatedOffers.getOfferIDList();
+
+      List<Resource> resources = new ArrayList<Resource>();
+      List<ResourceEntry> resourceEntryList = null;
+
+      try {
+        resourceEntryList = aggregatedOffers.reserveAndGet(ResourceType.CPU, new ScalarResourceEntry(logviewerCpu));
+        resources.addAll(createMesosScalarResourceList(ResourceType.CPU, resourceEntryList));
+        resourceEntryList = aggregatedOffers.reserveAndGet(ResourceType.MEM, new ScalarResourceEntry(logviewerMem));
+        resources.addAll(createMesosScalarResourceList(ResourceType.MEM, resourceEntryList));
+      } catch (ResourceNotAvailableException e) {
+        LOG.warn("launchLogviewer: Unable to launch logviewer because some resources were unavailable. Available aggregatedOffers: %s", aggregatedOffers);
+        continue;
+      }
+
+      CommandInfo.Builder commandInfoBuilder = CommandInfo.newBuilder()
+              .addUris(URI.newBuilder().setValue((String) mesosStormConf.get(CONF_EXECUTOR_URI)))
+              .addUris(URI.newBuilder().setValue(configUri))
+              .setValue(logviewerCommand);
+
+      TaskID taskId = TaskID.newBuilder()
+              .setValue(String.format("%s%slogviewer",  nodeId, DEFAULT_MESOS_COMPONENT_NAME_DELIMITER))
```

Comment:
e.g., of existing Task ID and Task Name for logviewer:
* Task ID: `worker-host4.dev|logviewer`
* Task Name: `logviewer`

And here's an existing Task ID and Task Name for a storm worker process:
* Task ID: `worker-host4.dev-31000-1503091419.181
* Task Name: `storm-worker | topology-foo-16-1493410791 | worker-host4.dev:31000`

Let's think about how to improve all of these for the situation where we're in a Blue-Green deployment situation with multiple Storm-on-Mesos frameworks running in the same Mesos cluster.  Goals:
1. Task ID *must* be unique or Mesos has major issues (see [this thread](https://mail-archives.apache.org/mod_mbox/mesos-user/201602.mbox/%3CCAO5KYW8%2BXMWc1dXtEo20BAsfGow028jwjL2ubMinP%2BK%2BvdOh8w@mail.gmail.com%3E) on the mesos user list where I posted an issue we encountered).
   * For worker processes that's handled by the combo of host + port, but also throwing in the timestamp of the launching of the task ensures it is unique.
   * For logviewer, it won't be unique if there is more than 1 Storm framework running.  We can make it unique by shoving in either the logviewer port and/or the Framework name.  More on this decision below.
2. Task Name should be meaningful.
   * For the worker process it's pretty useful (has the type of process, the topology ID and the slot ID), but it's lacking the framework name. e.g., `Storm v1.0.3 | storm-worker | ....`.   We can make that change with your PR or in a separate PR.
   * For the logviewer process it should at the minimum be enhanced to have:
        1. `storm-logviewer` instead of `logviewer`.
        2. ` | ` delimited strings with more information.
        3. The framework name.
        4. The logviewer port (from the storm config that is known to the MesosNimbus, it's `logviewer.port`).
    * So I'm thinking this would be good:  `<Framework Name> | storm-logviewer | port <logviewer.port>`.
       * e.g., `Storm v1.0.3 | storm-logviewer | port 8888`
3. Task ID appears in logs more than Task Name, so we should think about things we might wanna grep for in the logs.
   * e.g., Maybe it will be useful to grep for the framework name, so maybe we should be shoving that into mesos IDs everywhere (Executor IDs (logviewer & supervisor), Task IDs (logviewer and worker)).
      * *However*, I'm not sure if it's ok to have spaces in these Mesos IDs.  And the Framework Name very well could have spaces.